tests/components/bsblan/test_water_heater.py

```diff
@@ -0,0 +1,253 @@
+        )
+
+
+async def test_set_temperature(
+    hass: HomeAssistant,
+    mock_bsblan_with_methods: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setting temperature."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    await hass.services.async_call(
+        WATER_HEATER_DOMAIN,
+        "set_temperature",
+        {
+            ATTR_ENTITY_ID: ENTITY_ID,
+            ATTR_TEMPERATURE: 50,
+        },
+        blocking=True,
+    )
+
+    mock_bsblan_with_methods.set_hot_water.assert_called_once_with(nominal_setpoint=50)
+
+
+async def test_set_temperature_failure(
+    hass: HomeAssistant,
+    mock_bsblan_with_methods: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test setting temperature with API failure."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    mock_bsblan_with_methods.set_hot_water.side_effect = BSBLANError("Test error")
+
+    with pytest.raises(
+        HomeAssistantError, match="Failed to set target temperature for water heater"
+    ):
+        await hass.services.async_call(
+            WATER_HEATER_DOMAIN,
+            "set_temperature",
+            {
+                ATTR_ENTITY_ID: ENTITY_ID,
+                ATTR_TEMPERATURE: 50,
+            },
+            blocking=True,
+        )
+
+
+async def test_operation_mode_error(
+    hass: HomeAssistant,
+    mock_bsblan_with_methods: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test operation mode setting with API failure."""
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    mock_bsblan_with_methods.set_hot_water.side_effect = BSBLANError("Test error")
+
+    with pytest.raises(
+        HomeAssistantError, match="Failed to set operation mode for water heater"
+    ):
+        await hass.services.async_call(
+            WATER_HEATER_DOMAIN,
+            "set_operation_mode",
+            {
+                ATTR_ENTITY_ID: ENTITY_ID,
+                "operation_mode": STATE_ECO,
+            },
+            blocking=True,
+        )
+
+
+@pytest.mark.parametrize(
+    ("static_file"),
+    [
+        ("static.json"),
+        ("static_F.json"),
+    ],
+)
+async def test_temperature_unit(
+    hass: HomeAssistant,
+    mock_bsblan: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    snapshot: SnapshotAssertion,
+    entity_registry: er.EntityRegistry,
+    static_file: str,
+) -> None:
+    """Test Celsius and Fahrenheit temperature units."""
+
+    static_data = load_json_object_fixture(static_file, DOMAIN)
+
+    mock_bsblan.static_values.return_value = StaticState.from_dict(static_data)
+
+    await setup_with_selected_platforms(
+        hass, mock_config_entry, [Platform.WATER_HEATER]
+    )
+
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
```

Comment:
I am missing a bit what you want to test here since you already have a state test above. In this case, if the difference is only the unit of measurement, make a normal test, rather than a snapshot test